src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -264,34 +265,48 @@ public function supportsDenormalization(mixed $data, string $type, string $forma
     private function getNormalizer(mixed $data, ?string $format, array $context): ?NormalizerInterface
     {
         $type = \is_object($data) ? \get_class($data) : 'native-'.\gettype($data);
-
+        $minCached = CacheableSupport::SupportAlways;
+        $minUncached = CacheableSupport::SupportNever;
         if (!isset($this->normalizerCache[$format][$type])) {
+            $minCached = CacheableSupport::Support;
+            $minUncached = CacheableSupport::SupportNot;
             $this->normalizerCache[$format][$type] = [];
-
             foreach ($this->normalizers as $k => $normalizer) {
                 if (!$normalizer instanceof NormalizerInterface) {
                     continue;
                 }
 
-                if (!$normalizer instanceof CacheableSupportsMethodInterface || !$normalizer->hasCacheableSupportsMethod()) {
-                    $this->normalizerCache[$format][$type][$k] = false;
-                } elseif ($normalizer->supportsNormalization($data, $format, $context)) {
-                    $this->normalizerCache[$format][$type][$k] = true;
+                $support = $this->supportsNormalizationWrapper($normalizer, $data, $format, $context);
+                if (CacheableSupport::SupportNever === $support) {
+                    continue;
+                }
+
+                $this->normalizerCache[$format][$type][$k] = $support;
+                if (CacheableSupport::SupportAlways === $support) {
                     break;
                 }
             }
         }
 
         foreach ($this->normalizerCache[$format][$type] as $k => $cached) {
             $normalizer = $this->normalizers[$k];
-            if ($cached || $normalizer->supportsNormalization($data, $format, $context)) {
+            if ($cached->value >= $minCached->value || ($cached->value > $minUncached->value && $this->supportsNormalizationWrapper($normalizer, $data, $format, $context)->value > CacheableSupport::SupportNot->value)) {
```

Comment:
Just wondering, why is it different from the `getDenormalizer`? (I'm talking about the `minUncached` for example)
```
            if ($cached->value >= $minCached->value || ($cached->value > CacheableSupport::SupportNever->value && $this->supportsDenormalizationWrapper($normalizer, $data, $class, $format, $context)->value > CacheableSupport::SupportNot->value)) {
```

Reply:
I think they can be the same indeed